src/common/src/estimate_size/mod.rs

```diff
@@ -31,11 +32,8 @@ pub trait EstimateSize {
 
     /// The estimated total size of the current struct in bytes, including the `estimated_heap_size`
     /// and the size of `Self`.
-    fn estimated_size(&self) -> usize
-    where
-        Self: Sized,
-    {
-        self.estimated_heap_size() + std::mem::size_of::<Self>()
+    fn estimated_size(&self) -> usize {
+        self.estimated_heap_size() + std::mem::size_of_val(self)
```

Comment:
It should be the same for the same type.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=851f7ef5dfa625d080e1d2381bc83dfa

Reply:
cc. @BugenZhao

Reply:
The difference is we can now implement `EstimateSize` for dynamically sized types. Not sure whether this can be confusing... 🤔

<img width="545" alt="image" src="https://github.com/risingwavelabs/risingwave/assets/25862682/bf9b8275-fdc8-451a-babc-39be4f7b98e1">

Reply:
You are right. After reverting this change, the estimated size for Sum agg become correct (40 -> 32). 🥵